packages/workflow-apps-defaults/src/index.js

```diff
@@ -3,67 +3,111 @@
 
 import { baseFolder } from 'shared/env';
 import { join } from 'path';
+import defaultBrowser from 'default-browser';
 
 const defaultApps = ['Terminal', 'Browser', 'TextEditor'];
 
-let defaults = {};
-
-if (process.browser) {
-  defaults = require('workflow-apps-html').defaults;
-} else {
-  const platformDefaults = (() => {
-    switch (process.platform) {
-      case 'darwin':
-        return {
-          Terminal: require('workflow-app-iterm'),
-          Browser: require('workflow-app-safari'),
-          TextEditor: require('workflow-app-atom'),
-        };
-      case 'win32':
-        return {
-          Terminal: require('workflow-app-powershell'),
-          Browser: require('workflow-app-chrome'),
-          TextEditor: require('workflow-app-notepad'),
-        };
-      case 'linux':
-        return {
-          Terminal: require('workflow-app-xterm'),
-          Browser: require('workflow-app-chrome'),
-          TextEditor: require('workflow-app-atom'),
-        };
-      default:
-        console.log(`Platform '${process.platform}' not supported`);
-        console.log(
-          'Look for an issue for your platform here: https://github.com/havardh/workflow/issues'
-        );
-        process.exit(0);
-        break;
+async function getPlatformDefaults() {
+  switch (process.platform) {
+    case 'darwin': {
+      const browserName = (await defaultBrowser()).name.toLowerCase();
+      let Browser;
+      try {
+        Browser = require(`workflow-app-${browserName}`);
+      } catch (error) {
+        console.log(`Cannot find ${browserName}, falling back to safari`);
+        Browser = require('workflow-app-safari');
+      }
+      return {
+        Browser,
+        Terminal: require('workflow-app-iterm'),
+        TextEditor: require('workflow-app-atom'),
+      };
+    }
+    case 'win32':
+      return {
+        Terminal: require('workflow-app-powershell'),
+        Browser: require('workflow-app-chrome'),
+        TextEditor: require('workflow-app-notepad'),
+      };
+    case 'linux': {
+      const browserName = (await defaultBrowser()).name.toLowerCase();
```

Comment:
This returns `Google Chrome` my environment. And it is a bit unfortunate to depend on multiple external tools for the default names for packages and to expect that they behave portable. 

We need to add a mapping table for this so that multiple names can resolve to `workflow-app-chrome`. We should probably make it configurable so that new users could set a property to get it working and not have to wait for a new release. 

For configurability I suggest that we rely on the `command line arguments` (which you already proposed will be backed by environment variables, and i think `yargs` will give us a _rc_ file with only a single line of code). To do so I think we would need the `loader` function to be called with the `args` parameter passed to the `transformer`. It is a bit unobvious what to names these props though `--workflow-apps-defaults-workflow-app-chrome-platform-name` is a bit much. :thinking:

Reply:
> This returns Google Chrome my environment. And it is a bit unfortunate to depend on multiple external tools for the default names for packages and to expect that they behave portable. We need to add a mapping table for this so that multiple names can resolve to workflow-app-chrome.

👍 

> We should probably make it configurable so that new users could set a property to get it working and not have to wait for a new release.

Isn't this solved by configuring default in workflow settings as it will override the "platform default".